python/ray/rllib/trpo/trpo.py

```diff
@@ -0,0 +1,117 @@
+from __future__ import absolute_import, division, print_function
+
+import numpy as np
+import ray
+from ray.tune.trial import Resources
+from ray.rllib.agent import Agent
+from ray.rllib.optimizers import LocalSyncOptimizer
+from ray.tune.result import TrainingResult
+
+from ray.rllib.trpo.trpo_evaluator import TRPOEvaluator
+
+DEFAULT_CONFIG = {
+    # Number of workers (excluding master)
+    'num_workers': 4,
+    'use_gpu_for_workers': False,
+    'vf_loss_coeff': 0.5,
+    'use_lstm': False,
+    # Size of rollout batch
+    'batch_size': 512,
+    # Discount factor of MDP
+    'gamma': 0.99,
+    'use_pytorch': True,
+    'observation_filter': 'NoFilter',
+    'reward_filter': 'NoFilter',
+    'ent_coeff': 0.0,
+    'max_kl': 0.001,
+    'cg_damping': 0.001,
+    'residual_tol': 1e-10,
+
+    # Number of steps after which the rollout gets cut
+    'horizon': 500,
+    # Learning rate
+    'lr': 0.0004,
+    # Arguments to pass to the RLlib optimizer
+    'optimizer': {},
+    # Model parameters
+    'model': {
+        'fcnet_hiddens': [128, 128]
+    },
+    # Arguments to pass to the env creator
+    'env_config': {},
+}
+
+
+class TRPOAgent(Agent):
+    _agent_name = 'TRPO'
+    _default_config = DEFAULT_CONFIG
+    _allow_unknown_subkeys = ['model', 'optimizer', 'env_config']
+
+    @classmethod
+    def default_resource_request(cls, config):
+        cf = dict(cls._default_config, **config)
+        return Resources(
+            cpu=1,
+            gpu=0,
+            extra_cpu=cf["num_workers"],
+            extra_gpu=cf["use_gpu_for_workers"] and cf["num_workers"] or 0)
+
+    def _init(self):
+
+        self.local_evaluator = TRPOEvaluator(
+            self.registry,
+            self.env_creator,
+            self.config,
+        )
+
+        # TODO set weights fails for remote evaluators
```

Comment:
consider using a non-remote evaluator class to help debug.

Reply:
I actually fixed that particular bug today. It's because of https://github.com/ray-project/ray/issues/1856 and requires a custom serializer for torch.